### ISSUE TITLE
revive config schema rejects `enforce-switch-style` and `unsecure-url-scheme`, although both are documented

### ISSUE THREAD

**Issue by maratvk38** [golangci/golangci-lint#6475]
### Welcome

- [x] Yes, I'm using a binary release within 2 latest [releases](https://github.com/golangci/golangci-lint/releases). Only such installations are supported.
- [x] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [x] Yes, I've read the `typecheck` section of the [FAQ](https://golangci-lint.run/docs/welcome/faq/#why-do-you-have-typecheck-errors).
- [x] Yes, I've tried with the standalone [linter](https://golangci-lint.run/docs/linters/) if available (e.g., gocritic, go vet, etc.).
- [x] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### How did you install golangci-lint?

Other

### Description of the problem

`golangci-lint-config-verify` rejects some `revive` rules as invalid:

- `enforce-switch-style`
- `unsecure-url-scheme`

However, both rules are listed in the official `golangci-lint` documentation for `revive` configuration
https://golangci-lint.run/docs/linters/configuration/#revive

This looks like a mismatch between the documented supported rules and the JSON schema used by config validation

### Version of golangci-lint

<details>

```console
> head -n3 .pre-commit-config.yaml
repos:
  - repo: https://github.com/golangci/golangci-lint
    rev: v2.11.4
```

</details>

### Configuration

<details>

```console
version: "2"

run:
  tests: true
  timeout: 1m

linters:
  default: none

  enable:
#    - errcheck
    - govet
    - ineffassign
    - staticcheck
    - unused
    - asasalint
    - asciicheck
    - bidichk
    - bodyclose
    - copyloopvar
#    - dupl
    - durationcheck
    - errchkjson
    - errname
    - exhaustive
    - gocheckcompilerdirectives
#    - goconst
    - gosmopolitan
    - grouper
    - importas
    - lll
    - loggercheck
    - makezero
    - mirror
    - musttag
    - nilerr
    - nilnil
    - noctx
    - nolintlint
    - prealloc
    - predeclared
    - reassign
    - revive
    - sloglint
    - sqlclosecheck
    - tagalign
    - testifylint
    - tparallel
    - unconvert
    - unparam
    - usestdlibvars
    - wastedassign
    - whitespace

  settings:
    errcheck:
      check-type-assertions: false
      check-blank: false

    goconst:
      min-len: 2
      min-occurrences: 4

    govet:
      enable-all: true
      disable:
        - fieldalignment
        - shadow
        - nilness

    lll:
      line-length: 140
      tab-width: 1

    importas:
      no-extra-aliases: false

    staticcheck:
      checks:
        - all
        - -ST1000
        - -ST1003
        - -ST1020
        - -ST1021
        - -ST1022

    revive:
      severity: warning
      confidence: 0.8
      enable-all-rules: true
      rules:
        - name: package-comments
          disabled: true

        - name: exported
          disabled: true

        - name: nested-structs
          disabled: true

        - name: line-length-limit
          disabled: true

        - name: add-constant
          disabled: true

        - name: max-public-structs
          disabled: true

        - name: import-alias-naming
          disabled: true

        - name: flag-parameter
          disabled: true

        - name: unsecure-url-scheme
          disabled: true

        - name: enforce-switch-style
          disabled: true

        - name: import-shadowing
          disabled: true

        - name: unchecked-type-assertion
          disabled: true

        - name: unexported-naming
          disabled: true

        - name: unhandled-error
          disabled: true

        - name: cyclomatic
          disabled: true

        - name: cognitive-complexity
          disabled: true

        - name: function-length
          disabled: true

        - name: package-directory-mismatch
          disabled: true

        - name: unused-receiver
          disabled: true

        - name: confusing-results
          disabled: true

        - name: package-naming
          disabled: true

        - name: struct-tag
          arguments:
            - validate,cluster
            - validate,duration
            - validate,severity

        - name: max-control-nesting
          arguments: [7]

        - name: function-result-limit
          arguments: [5]

        - name: argument-limit
          arguments: [16]

    prealloc:
      simple: true
      range-loops: true
      for-loops: false

    whitespace:
      multi-if: false
      multi-func: false

  exclusions:
    generated: lax
    presets:
      - comments
      - common-false-positives
      - legacy
      - std-error-handling

    rules:
      - path: _test\.go
        linters:
          - gocyclo
          - errcheck
          - dupl
          - gosec
          - staticcheck
          - testifylint
          - lll

      - linters:
          - lll
        source: "^//go:generate "

    paths:
      - perfomance_test
      - environments

formatters:
  enable:
    - gofumpt
    - golines
  settings:
    golines:
      max-len: 140 # based on lll

issues:
  max-issues-per-linter: 0
  max-same-issues: 0
```

</details>

### Go environment

<details>

```console
> go version && go env
go version go1.25.7 darwin/arm64
AR='ar'
CC='clang'
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_ENABLED='1'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
CXX='clang++'
GCCGO='gccgo'
GO111MODULE='on'
GOARCH='arm64'
GOARM64='v8.0'
GOAUTH='netrc'
GOBIN=''
GOCACHE='/Users/<private>/Library/Caches/go-build'
GOCACHEPROG=''
GODEBUG=''
GOENV='/Users/<private>/Library/Application Support/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFIPS140='off'
GOFLAGS=''
GOGCCFLAGS='-fPIC -arch arm64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -ffile-prefix-map=/var/folders/7_/bmcyv6ls11n0tm0wtwxgyc280000gp/T/go-build2772575355=/tmp/go-build -gno-record-gcc-switches -fno-common'
GOHOSTARCH='arm64'
GOHOSTOS='darwin'
GOINSECURE=''
GOMOD='/Users/<private>/projects/<private>/<private>/go.mod'
GOMODCACHE='/Users/<private>/go/pkg/mod'
GONOPROXY=''
GONOSUMDB='gitlab.<private>'
GOOS='darwin'
GOPATH='/Users/<private>/go'
GOPRIVATE=''
GOPROXY='https://<private>/repository/golang,https://proxy.golang.org,direct'
GOROOT='/Users/<private>/.goenv/versions/1.25.7'
GOSUMDB='sum.golang.org'
GOTELEMETRY='local'
GOTELEMETRYDIR='/Users/<private>/Library/Application Support/go/telemetry'
GOTMPDIR=''
GOTOOLCHAIN='auto'
GOTOOLDIR='/Users/<private>/.goenv/versions/1.25.7/pkg/tool/darwin_arm64'
GOVCS=''
GOVERSION='go1.25.7'
GOWORK=''
PKG_CONFIG='pkg-config'
```

</details>

### Verbose output of running

<details>

```console
> prek run --all-files
golangci-lint-config-verify..............................................Failed
- hook id: golangci-lint-config-verify
- exit code: 3

  jsonschema: "linters.settings.revive.rules.8.name" does not validate with "/properties/linters/properties/settings/properties/revive/$ref/properties/rules/items/properties/name/$ref/enum": value must be one of 'add-constant', 'argument-limit', 'atomic', 'banned-characters', 'bare-return', 'blank-imports', 'bool-literal-in-expr', 'call-to-gc', 'cognitive-complexity', 'comment-spacings', 'comments-density', 'confusing-naming', 'confusing-results', 'constant-logical-expr', 'context-as-argument', 'context-keys-type', 'cyclomatic', 'datarace', 'deep-exit', 'defer', 'dot-imports', 'duplicated-imports', 'early-return', 'empty-block', 'empty-lines', 'enforce-map-style', 'enforce-repeated-arg-type-style', 'enforce-slice-style', 'error-naming', 'error-return', 'error-strings', 'errorf', 'exported', 'file-header', 'file-length-limit', 'filename-format', 'flag-parameter', 'function-length', 'function-result-limit', 'get-return', 'identical-branches', 'if-return', 'import-alias-naming', 'import-shadowing', 'imports-blocklist', 'increment-decrement', 'indent-error-flow', 'line-length-limit', 'max-control-nesting', 'max-public-structs', 'modifies-parameter', 'modifies-value-receiver', 'nested-structs', 'optimize-operands-order', 'package-comments', 'range-val-address', 'range-val-in-closure', 'range', 'receiver-naming', 'redefines-builtin-id', 'redundant-build-tag', 'redundant-import-alias', 'redundant-test-main-exit', 'string-format', 'string-of-int', 'struct-tag', 'superfluous-else', 'time-equal', 'time-naming', 'unchecked-type-assertion', 'unconditional-recursion', 'unexported-naming', 'unexported-return', 'unhandled-error', 'unnecessary-stmt', 'unreachable-code', 'unused-parameter', 'unused-receiver', 'use-any', 'use-errors-new', 'useless-break', 'var-declaration', 'var-naming', 'waitgroup-by-value'
  jsonschema: "linters.settings.revive.rules.9.name" does not validate with "/properties/linters/properties/settings/properties/revive/$ref/properties/rules/items/properties/name/$ref/enum": value must be one of 'add-constant', 'argument-limit', 'atomic', 'banned-characters', 'bare-return', 'blank-imports', 'bool-literal-in-expr', 'call-to-gc', 'cognitive-complexity', 'comment-spacings', 'comments-density', 'confusing-naming', 'confusing-results', 'constant-logical-expr', 'context-as-argument', 'context-keys-type', 'cyclomatic', 'datarace', 'deep-exit', 'defer', 'dot-imports', 'duplicated-imports', 'early-return', 'empty-block', 'empty-lines', 'enforce-map-style', 'enforce-repeated-arg-type-style', 'enforce-slice-style', 'error-naming', 'error-return', 'error-strings', 'errorf', 'exported', 'file-header', 'file-length-limit', 'filename-format', 'flag-parameter', 'function-length', 'function-result-limit', 'get-return', 'identical-branches', 'if-return', 'import-alias-naming', 'import-shadowing', 'imports-blocklist', 'increment-decrement', 'indent-error-flow', 'line-length-limit', 'max-control-nesting', 'max-public-structs', 'modifies-parameter', 'modifies-value-receiver', 'nested-structs', 'optimize-operands-order', 'package-comments', 'range-val-address', 'range-val-in-closure', 'range', 'receiver-naming', 'redefines-builtin-id', 'redundant-build-tag', 'redundant-import-alias', 'redundant-test-main-exit', 'string-format', 'string-of-int', 'struct-tag', 'superfluous-else', 'time-equal', 'time-naming', 'unchecked-type-assertion', 'unconditional-recursion', 'unexported-naming', 'unexported-return', 'unhandled-error', 'unnecessary-stmt', 'unreachable-code', 'unused-parameter', 'unused-receiver', 'use-any', 'use-errors-new', 'useless-break', 'var-declaration', 'var-naming', 'waitgroup-by-value'
  jsonschema: "linters.settings.revive.rules.17.name" does not validate with "/properties/linters/properties/settings/properties/revive/$ref/properties/rules/items/properties/name/$ref/enum": value must be one of 'add-constant', 'argument-limit', 'atomic', 'banned-characters', 'bare-return', 'blank-imports', 'bool-literal-in-expr', 'call-to-gc', 'cognitive-complexity', 'comment-spacings', 'comments-density', 'confusing-naming', 'confusing-results', 'constant-logical-expr', 'context-as-argument', 'context-keys-type', 'cyclomatic', 'datarace', 'deep-exit', 'defer', 'dot-imports', 'duplicated-imports', 'early-return', 'empty-block', 'empty-lines', 'enforce-map-style', 'enforce-repeated-arg-type-style', 'enforce-slice-style', 'error-naming', 'error-return', 'error-strings', 'errorf', 'exported', 'file-header', 'file-length-limit', 'filename-format', 'flag-parameter', 'function-length', 'function-result-limit', 'get-return', 'identical-branches', 'if-return', 'import-alias-naming', 'import-shadowing', 'imports-blocklist', 'increment-decrement', 'indent-error-flow', 'line-length-limit', 'max-control-nesting', 'max-public-structs', 'modifies-parameter', 'modifies-value-receiver', 'nested-structs', 'optimize-operands-order', 'package-comments', 'range-val-address', 'range-val-in-closure', 'range', 'receiver-naming', 'redefines-builtin-id', 'redundant-build-tag', 'redundant-import-alias', 'redundant-test-main-exit', 'string-format', 'string-of-int', 'struct-tag', 'superfluous-else', 'time-equal', 'time-naming', 'unchecked-type-assertion', 'unconditional-recursion', 'unexported-naming', 'unexported-return', 'unhandled-error', 'unnecessary-stmt', 'unreachable-code', 'unused-parameter', 'unused-receiver', 'use-any', 'use-errors-new', 'useless-break', 'var-declaration', 'var-naming', 'waitgroup-by-value'
  jsonschema: "linters.settings.revive.rules.20.name" does not validate with "/properties/linters/properties/settings/properties/revive/$ref/properties/rules/items/properties/name/$ref/enum": value must be one of 'add-constant', 'argument-limit', 'atomic', 'banned-characters', 'bare-return', 'blank-imports', 'bool-literal-in-expr', 'call-to-gc', 'cognitive-complexity', 'comment-spacings', 'comments-density', 'confusing-naming', 'confusing-results', 'constant-logical-expr', 'context-as-argument', 'context-keys-type', 'cyclomatic', 'datarace', 'deep-exit', 'defer', 'dot-imports', 'duplicated-imports', 'early-return', 'empty-block', 'empty-lines', 'enforce-map-style', 'enforce-repeated-arg-type-style', 'enforce-slice-style', 'error-naming', 'error-return', 'error-strings', 'errorf', 'exported', 'file-header', 'file-length-limit', 'filename-format', 'flag-parameter', 'function-length', 'function-result-limit', 'get-return', 'identical-branches', 'if-return', 'import-alias-naming', 'import-shadowing', 'imports-blocklist', 'increment-decrement', 'indent-error-flow', 'line-length-limit', 'max-control-nesting', 'max-public-structs', 'modifies-parameter', 'modifies-value-receiver', 'nested-structs', 'optimize-operands-order', 'package-comments', 'range-val-address', 'range-val-in-closure', 'range', 'receiver-naming', 'redefines-builtin-id', 'redundant-build-tag', 'redundant-import-alias', 'redundant-test-main-exit', 'string-format', 'string-of-int', 'struct-tag', 'superfluous-else', 'time-equal', 'time-naming', 'unchecked-type-assertion', 'unconditional-recursion', 'unexported-naming', 'unexported-return', 'unhandled-error', 'unnecessary-stmt', 'unreachable-code', 'unused-parameter', 'unused-receiver', 'use-any', 'use-errors-new', 'useless-break', 'var-declaration', 'var-naming', 'waitgroup-by-value'
  The command is terminated due to an error: the configuration contains invalid elements
golangci-lint-full.......................................................Passed
```

</details>

### A minimal reproducible example or link to a public repository

<details>

Minimal config:

```yaml
linters:
  settings:
    revive:
      rules:
        - name: enforce-switch-style
        - name: unsecure-url-scheme
```
`.pre-commit-config.yaml`

```yaml
repos:
  - repo: https://github.com/golangci/golangci-lint
    rev: v2.11.4
    hooks:
      - id: golangci-lint-config-verify
      - id: golangci-lint-full
```

</details>

### Validation

- [x] Yes, I've included all information above (version, config, etc.).

### Supporter

- [ ] I am a [sponsor/backer](https://donate.golangci.org) of this project.

**Comment by ldez** [comment]
Duplicate of #6347
